File: ChattersApp/src/components/ConnectionBanner.js
import { StyleSheet, Text, View } from "react-native";

import { useTheme } from "../theme";

/**
 * Surfaces socket state so a dropped connection is visible instead of the app
 * silently ceasing to receive messages.
 */
export default function ConnectionBanner({ status }) {
  const theme = useTheme();
  if (status === "online") return null;

  const label =
    status === "connecting"
      ? "Connecting…"
      : status === "reconnecting"
      ? "Reconnecting…"
      : "Offline — messages will sync when you reconnect";

  return (
    <View style={[styles.banner, { backgroundColor: theme.warnBg }]}>
      <Text style={[styles.text, { color: theme.warnText }]}>{label}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  banner: { paddingVertical: 6, paddingHorizontal: 12 },
  text: { fontSize: 12, textAlign: "center" },
});

File: flutter/chatters_shared/lib/src/ui/widgets/connection_banner.dart
import 'package:flutter/material.dart';

import '../../services/chat_socket.dart';
import '../theme.dart';

class ConnectionBanner extends StatelessWidget {
  const ConnectionBanner({super.key, required this.status});
  final SocketStatus status;

  @override
  Widget build(BuildContext context) {
    if (status == SocketStatus.online) return const SizedBox.shrink();
    final label = switch (status) {
      SocketStatus.connecting => 'Connecting…',
      SocketStatus.reconnecting => 'Reconnecting…',
      _ => 'Offline',
    };
    return Container(
      width: double.infinity,
      color: context.p.warnBg,
      padding: const EdgeInsets.symmetric(vertical: 6),
      child: Text(label, textAlign: TextAlign.center, style: TextStyle(color: context.p.warnText, fontSize: 12)),
    );
  }
}
